### ISSUE TITLE
fix: Empty Href behavior not consistent between NavLink and FluentNavLink

### ISSUE THREAD

**Issue by Hwiet** [microsoft/fluentui-blazor#1654]
# 🐛 Bug Report

`FluentNavLink` does not create a link when the Href parameter is empty.

This is different from the expected behavior of `NavLink` component that comes with the Blazor framework, where an empty Href creates a link to the root of the application. In an unmodified _Blazor WebAssembly Standalone App_ template available on Visual Studio, this is how the link to the root of the application is written in `NavMenu.razor`:

```html
 <div class="nav-item px-3">
     <NavLink class="nav-link" href="" Match="NavLinkMatch.All">
         <span class="bi bi-house-door-fill-nav-menu" aria-hidden="true"></span> Home
     </NavLink>
 </div>
 <div class="nav-item px-3">
     <NavLink class="nav-link" href="counter">
         <span class="bi bi-plus-square-fill-nav-menu" aria-hidden="true"></span> Counter
     </NavLink>
 </div>
```

...the link does not work anymore after switching `NavLink` to `FluentNavLink`.

This is not a big issue as passing `./` to the `Href` parameter seems to achieve the same result but it would be better if the behavior of `FluentNavLink` matches that of `NavLink`.

## 🤔 Expected Behavior

An empty string passed to the `Href` parameter of `FluentNavLink` should create a link to the root of the Blazor application.

## 😯 Current Behavior

An empty string passed to the `Href` parameter of `FluentNavLink` does not create a link at all.

## 💁 Possible Solution

Remove the `!string.IsNullOrEmpty(Href)` condition on [FluentNavLink.razor](https://github.com/microsoft/fluentui-blazor/blob/89004de943779ff8be71a589341c6805a024c21a/src/Core/Components/NavMenu/FluentNavLink.razor#L7)

**Comment by vnbaaij** [comment]
This is done on purpose so you can have entries in the FluentNavMenu that don't necessarily go somewhere but do invoke an action. Like in the https://www.fluentui-blazor.net/NavMenu#navigation-examples:

'``
<FluentNavLink OnClick="OnClick" Icon="@(new Icons.Regular.Size20.LeafOne())">Item 3.1</FluentNavLink>
```

**Comment by peterharding93** [comment]
I have come across this problem.  I can't seem to find a workaround when working behind a proxy with a PathBase.  I want the link to home page to work which I have previosly been rendering as href="", but this no longer works.  I can't pass Href="/" because the browser will not intrepret it as being relative to the PathBase.  
  Is there an intended workaround, or was this scenario not considered?  I image this is a very common scenario.
My recommendation: accept empty string as valid href, but keep the current intrepretation of a null value as it is.

**Comment by Hwiet** [comment]
Mentioned in the issue, but passing `"./"` instead of an empty string should work.

**Comment by peterharding93** [comment]
> Mentioned in the issue, but passing `"./"` instead of an empty string should work.
Yes, you are right.  I thought that wasn't working either, but after testing again and properly, it seems to work and is probably the correct solution.  Thanks.
